app/referee/job_submitter.py

```diff
@@ -73,25 +71,28 @@ def check_for_submission_runs(start_time, client, check_interval, job_lifetime):
     while job_submitter_alive(start_time, check_interval, job_lifetime):
         logger.debug("Checking for submission runs n=%d", n)
         with transaction.atomic():
-            for run in SubmissionRun.objects.select_for_update().filter(
-                status=Status.PENDING_REMOTE
-            ):
+            submission_run_query = (
+                SubmissionRun.objects.select_for_update()
+                .filter(status=Status.PENDING_REMOTE)
+                .all()
```

Comment:
this isn't necessary, the query will run as soon as you start iterating through results